Updates plugin for TinyMCE Upgrade

sdfippinger opened the pull request:
TinyMCE upgrade from 4 to 6 requires a good bit of changes to this plugin. Also cleans up some code formatting to what we use now.

This should stay in Draft until we get the TinyMCE Update merged into master branch of SAM. Otherwise SAM may point to this v6 compatible plugin that will be broke in v4.

sdfippinger committed:
Updates plugin for TinyMCE Upgrade

TinyMCE upgrade from 4 to 6 requires a good bit of changes to this plugin.
Also cleans up some code formatting to what we use now.

sntran commented on package.json:
```diff
@@ -4,8 +4,8 @@
   "scripts": {
-    "build": "node_modules/.bin/uglifyjs --compress properties,dead_code --mangle --output plugin.min.js --source-map plugin.js.map plugin.js",
-    "docs": "cp -r plugin.js.map plugin.min.js img docs/scripts/tinymce/plugins/video",
+    "build": "node_modules/.bin/uglifyjs --compress properties,dead_code --mangle --source-map --output plugin.min.js plugin.js",
```
May as well switch to esbuild for these.

sntran commented on plugin.js:
```diff
@@ -45,60 +45,274 @@
+    this.configs = {};
+
+    this.configs["unknown"] = {
+      title: "Insert a video",
```
There are a lot of duplication here. We may want to have a single object for config, and for each providers, use spread operator to extend/override settings that are specific to them.

sntran reviewed (changes requested):
I suggested a couple improvements, and assumed the plugin works, since the diff are so big to compare.

sdfippinger committed:
refactor: Reduces repeated config settings

Updates build to use esbuild.
Build will now wire up the /docs/ demo site for development.
Uses the latest TinyMCE for the demo site.

sdfippinger committed:
Including index.js & setting version
